### PR TITLE
MNT: Bump matplotlib minversion to 3.2

### LINKED ISSUE #8832: MNT: Revert PR 8818 when minversion of matplotlib is 3.2
In `astropy/visualization/units.py`, as added in #8818 

```python
    # NOTE: in matplotlib >=3.2, subclasses will be recognized automatically,
    # and once that becomes our minimum version, we can remove this,
    # adding just u.Quantity itself to the registry.
```

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to bump minversion of matplotlib to 3.2 and this able to remove the now unnecessary code added in

* #8782
* #8818

Fixes #8832

### After merge

Add `py39-test-image-mpl322-cov` job to branch protection rule for `main` branch.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/visualization/tests/test_norm.py [hunk 1]
```diff
@@ -4,18 +4,12 @@
...
 
 from astropy.utils.compat.optional_deps import HAS_MATPLOTLIB, HAS_PLT
 from astropy.visualization.interval import ManualInterval, PercentileInterval
 from astropy.visualization.mpl_normalize import ImageNormalize, imshow_norm, simple_norm
 from astropy.visualization.stretch import LogStretch, PowerStretch, SqrtStretch
 
-if HAS_MATPLOTLIB:
-    import matplotlib
-
-    MATPLOTLIB_LT_32 = Version(matplotlib.__version__) < Version("3.2")
```
Comment (mhvk) [inline astropy/visualization/tests/test_norm.py:17]: Funny that this wasn't even used in the tests!

**Review by mhvk — COMMENTED** [review]
Thanks, this looks good but CI is erroring.

**Review by larrybradley — APPROVED** [review]
visualization changes look fine.  Not sure why the mpl image test failed.

**Comment by pllim** [comment]
I think that is because there is no baseline for `mpl322` config. But according to https://docs.astropy.org/en/latest/development/testguide.html#generating-reference-images , there is nothing I need to do? Things will just magically pass when this is merged?

**Comment by astrofrog** [comment]
I need to double check, I think we need to copy the hash file to this PR - will confirm tomorrow

**Comment by astrofrog** [comment]
@pllim - see https://github.com/astropy/astropy/pull/14329, you'll need to generate the hash library locally (or copy a previous one)

**Comment by pllim** [comment]
Does the failure log look normal? I cannot tell... https://app.circleci.com/pipelines/github/astropy/astropy/14299/workflows/3c0ffa75-a3a2-4f99-a213-0b6f9dcab803/jobs/108267

**Comment by pllim** [comment]
@astrofrog , you should look at the diff, some look suspicious to me... https://output.circle-artifacts.com/output/job/1f033492-7a62-4c8d-9720-f1c20e714941/artifacts/0/results/fig_comparison.html

**Comment by astrofrog** [comment]
The failures look ok to me, so I think you can safely update the hash file to the one you can download from the figure comparison page.

**Comment by pllim** [comment]
pre-commit autofix

🙄

**Comment by pllim** [comment]
pre-commit.ci autofix

**Comment by pllim** [comment]
Thanks, all!

**Pull request merged**